### PR TITLE
Number conversions(Binary to decimal)

### LINKED ISSUE #865: Number Convertions in different languages
Can i work on it ??

### PR DESCRIPTION
**Pull request by khushal87**
Fixes #865  

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:

- added binary to decimal conversions in c++,java,python,js
- 

#### Languages Used:
- c++
- java
- python
- javascript

#### Files Added:
- Binary_To_Decimal.cpp
- Binary_To_Decimal.java
- Binary_To_Decimal.js
- Binary_To_Decimal.py

### CODE REVIEW

File: Number_Conversions/Binary_To_Decimal/Binary_To_Decimal.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// Java program to convert binary to decimal 
+
+// Function to convert binary to decimal 
+class number { 
+	static int binaryToDecimal(int n)
```
Comment (ayushin78) [inline Number_Conversions/Binary_To_Decimal/Binary_To_Decimal.java]: Does this program work with this class name?

Comment (khushal87) [reply Number_Conversions/Binary_To_Decimal/Binary_To_Decimal.java]: yes

Comment (khushal87) [reply Number_Conversions/Binary_To_Decimal/Binary_To_Decimal.java]: its working

**Comment by khushal87** [comment]
> 
> 
> Can you modify it to take input from the user? Use string variable to store the input to make it work for long binary numbers as well.

I tried keeping it as simple with long binary number the desired decimal representation may get hampered because of the modulus and quotient

**Comment by jainaman224** [comment]
Please use 4 space indentation in all files.

File: Number_Conversions/Binary_To_Decimal/Binary_To_Decimal.cpp [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	int temp = num; 
+	while (temp) { 
+		int last_digit = temp % 10; 
+		temp = temp / 10; 
+		decimal_value += last_digit * base; 
+		base = base * 2; 
+	} 
+	return decimal_value; 
+} 
+//The main function
```
Comment (jainaman224) [inline Number_Conversions/Binary_To_Decimal/Binary_To_Decimal.cpp]: Add a new line above this.

File: Number_Conversions/Binary_To_Decimal/Binary_To_Decimal.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+		int temp = num; 
+		while (temp > 0) { 
+			int last_digit = temp % 10; 
+			temp = temp / 10; 
+			decimal_value += last_digit * base; 
+			base = base * 2; 
+		} 
+		return decimal_value; 
+	} 
+	// The main Function
```
Comment (jainaman224) [inline Number_Conversions/Binary_To_Decimal/Binary_To_Decimal.java]: Add a new line above this.

File: Number_Conversions/Binary_To_Decimal/Binary_To_Decimal.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    while (temp > 0) {
+        var last_digit = temp % 10;
+        var digitValue = last_digit * base;
+        decimal_value += digitValue;
+        base = base * 2;
+        temp = Math.floor(temp / 10);
+    }
+    return decimal_value;
+}
+// The main function
```
Comment (jainaman224) [inline Number_Conversions/Binary_To_Decimal/Binary_To_Decimal.js]: Add a new line above this.

File: Number_Conversions/Binary_To_Decimal/Binary_To_Decimal.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        temp = Math.floor(temp / 10);
+    }
+    return decimal_value;
+}
+// The main function 
+function main() {
+    var num = 101011101;
+    console.log(binaryToDecimal(num));
+}
+main();
```
Comment (jainaman224) [inline Number_Conversions/Binary_To_Decimal/Binary_To_Decimal.js]: Add a new line above this.

File: Number_Conversions/Binary_To_Decimal/Binary_To_Decimal.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
+// Java program to convert binary to decimal 
+
+// Function to convert binary to decimal 
+class number {
```
Comment (jainaman224) [inline Number_Conversions/Binary_To_Decimal/Binary_To_Decimal.java]: Please fix class name. This file won't compile.

**Comment by khushal87** [comment]
All requested changes are done

**Comment by jainaman224** [comment]
Use 4 space indentation instead of 8. It is not done.

File: Number_Conversions/Binary_To_Decimal/Binary_To_Decimal.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
+// Java program to convert binary to decimal 
+
+// Function to convert binary to decimal 
+class numberconversion { 
+	static int binaryToDecimal(int n)
```
Comment (jainaman224) [inline Number_Conversions/Binary_To_Decimal/Binary_To_Decimal.java]: How are you testing whether it is working or not?

Comment (khushal87) [reply Number_Conversions/Binary_To_Decimal/Binary_To_Decimal.java]: On  online ide its working

**Comment by jainaman224** [comment]
Use 4 space indentation instead of 8. It is not done.

**Comment by khushal87** [comment]
So what must i do ? To push and squash my commits

**Comment by jainaman224** [comment]
@khushal87 Indent your files with 4 spaces first. When your PR is approved, squash commits and push.

**Comment by khushal87** [comment]
> 
> 
> @khushal87 Indent your files with 4 spaces first. When your PR is approved, squash commits and push.

I have indented all the files now but unable to squash after many attempts and seeking help

**Comment by jainaman224** [comment]
You need to squash last 20 commits.
Run `git rebase -i HEAD~2`.
Replace `pick` with `squash` or `s` for 19 commits. It will squash them to first commit. Then write appropriate commit message. It will show you commit message for all 20 commits in a single file.
Exit editor and done.

Suggestion: To experiment create a new branch from this. If succeeds do the same in this branch.

File: Number_Conversions/Binary_To_Decimal/Binary_To_Decimal.cpp [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+// Function to convert binary to decimal 
+int binaryToDecimal(int n) 
+{ 
+    int decimal_value = 0; 
+
+    // Initializing base value to 1, i.e 2^0 
+    int base = 1; 
+    int temp = n; 
+    while (temp) 
+	{
```
Comment (jainaman224) [inline Number_Conversions/Binary_To_Decimal/Binary_To_Decimal.cpp]: Indentation needs to be fixed

File: Number_Conversions/Binary_To_Decimal/Binary_To_Decimal.cpp [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+	    decimal_value += last_digit * base; 
+	    base = base * 2; 
+    } 
+    return decimal_value; 
+} 
+
+//The main function 
+int main() 
+{ 
+	int num;
```
Comment (jainaman224) [inline Number_Conversions/Binary_To_Decimal/Binary_To_Decimal.cpp]: Indentation fix.

File: Number_Conversions/Binary_To_Decimal/Binary_To_Decimal.cpp [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+    cin>>num;
```
Comment (jainaman224) [inline Number_Conversions/Binary_To_Decimal/Binary_To_Decimal.cpp]: Space around `>>` would be great.

File: Number_Conversions/Binary_To_Decimal/Binary_To_Decimal.java [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+			decimal_value += last_digit * base; 
+			base = base * 2; 
+		} 
+		return decimal_value; 
+	} 
+	
+	// The main Function
+	public static void main(String[] args) 
+	{ 
+		int num = 101011101;
```
Comment (jainaman224) [inline Number_Conversions/Binary_To_Decimal/Binary_To_Decimal.java]: Indentation fix. Needs to be done thought out the file.

File: Number_Conversions/Binary_To_Decimal/Binary_To_Decimal.cpp (hunk 8 above)
Comment (jainaman224) [reply Number_Conversions/Binary_To_Decimal/Binary_To_Decimal.cpp]: Not fixed yet.

**Comment by jainaman224** [comment]
@khushal87 Are you working on it?

**Pull request closed without merging**